src/api/QdmCalculationService.test.ts

```diff
@@ -1158,6 +1159,37 @@ describe("QDM CalculationService Tests", () => {
     });
   });
 
+  it("should return testCase with actual values passing for patientBasis Ratio", () => {
+    const output = calculationService.processTestCaseResults(
```

Comment:
this looks more like an integration test which is good but I think it would be nice to add some unit tests for `mapObservations` to cover all if elses. This test will cover only couple of them.

Reply:
I have added test for the  mapPatientBasedObservations